### PR TITLE
Integrate GitHub's Stale bot for PRs and Issues

### PR DESCRIPTION
**Pull request by tyleroconnell**
Given the desire to more easily maintain this repo, I thought I'd bring in GitHub's [probot/stale](https://github.com/probot/stale) to automatically close PRs and issues that have not been addressed in the last 60 days. I have left all of the configuration items, in case there are some that we'd like to keep that I hadn't initially considered.

### CODE REVIEW

File: .github/stale.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Number of days of inactivity before an Issue or Pull Request becomes stale
+daysUntilStale: 60
+
+# Number of days of inactivity before an Issue or Pull Request with the stale label is closed.
+# Set to false to disable. If disabled, issues still need to be closed manually, but will remain marked as stale.
+daysUntilClose: 7
+
+# Only issues or pull requests with all of these labels are check if stale. Defaults to `[]` (disabled)
+onlyLabels: []
+
```
Comment (tyleroconnell) [inline .github/stale.yml:12]: I believe we'd like all issues/PRs to be considered
```suggestion
```

File: .github/stale.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+# Issues or Pull Requests with these labels will never be considered stale. Set to `[]` to disable
+exemptLabels:
+  - pinned
+  - security
+  - later
```
Comment (tyleroconnell) [inline .github/stale.yml:17]: I've changed `[Status] Maybe Later` to `later` for brevity

File: .github/stale.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+# Set to true to ignore issues in a project (defaults to false)
+exemptProjects: false
+
```
Comment (tyleroconnell) [inline .github/stale.yml:21]: I believe we would like all issues to be considered
```suggestion
```

File: .github/stale.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+# Set to true to ignore issues in a milestone (defaults to false)
+exemptMilestones: false
+
```
Comment (tyleroconnell) [inline .github/stale.yml:24]: I believe we would like all issues to be considered
```suggestion
```

File: .github/stale.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+# Set to true to ignore issues with an assignee (defaults to false)
+exemptAssignees: false
+
```
Comment (tyleroconnell) [inline .github/stale.yml:27]: I think it'd be good for stale to remind us about issues, even those already assigned
```suggestion
```

File: .github/stale.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+# Comment to post when marking as stale. Set to `false` to disable
+markComment: >
+  This issue has been automatically marked as stale because it has not had
+  recent activity. It will be closed if no further activity occurs. Thank you
+  for your contributions.
+
+# Comment to post when removing the stale label.
+# unmarkComment: >
+#   Your comment here.
+
```
Comment (tyleroconnell) [inline .github/stale.yml:40]: I don't think we need this
```suggestion
```

File: .github/stale.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+# Comment to post when closing a stale Issue or Pull Request.
+# closeComment: >
+#   Your comment here.
+
```
Comment (tyleroconnell) [inline .github/stale.yml:44]: I don't think we need this
```suggestion
```

File: .github/stale.yml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+# Limit the number of actions per hour, from 1-30. Default is 30
+limitPerRun: 30
+
```
Comment (tyleroconnell) [inline .github/stale.yml:47]: I think it's nice to have this called out explicitly

File: .github/stale.yml [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+# Limit to only `issues` or `pulls`
+# only: issues
+
```
Comment (tyleroconnell) [inline .github/stale.yml:50]: I believe we'd like all issues/PRs to be considered
```suggestion
```

File: .github/stale.yml [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+# Optionally, specify configuration settings that are specific to just 'issues' or 'pulls':
+# pulls:
+#   daysUntilStale: 30
+#   markComment: >
+#     This pull request has been automatically marked as stale because it has not had
+#     recent activity. It will be closed if no further activity occurs. Thank you
+#     for your contributions.
```
Comment (tyleroconnell) [inline .github/stale.yml:57]: I'm fine with treating issues and PRs the same
```suggestion
```

File: .github/stale.yml [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+# issues:
+#   exemptLabels:
+#     - confirmed
```
Comment (tyleroconnell) [inline .github/stale.yml:61]: I don't think we need this, at least not yet
```suggestion
```

**Review by tyleroconnell — COMMENTED** [review]
All values are their defaults, unless otherwise noted

File: .github/stale.yml (hunk 1 above)
Comment (ckstettler) [reply .github/stale.yml:12]: I agree.

File: .github/stale.yml (hunk 2 above)
Comment (ckstettler) [reply .github/stale.yml:17]: These look reasonable.

File: .github/stale.yml (hunk 3 above)
Comment (ckstettler) [reply .github/stale.yml:21]: Yes.

File: .github/stale.yml (hunk 5 above)
Comment (ckstettler) [reply .github/stale.yml:27]: Seems good.

File: .github/stale.yml (hunk 4 above)
Comment (ckstettler) [reply .github/stale.yml:24]: seems good.

File: .github/stale.yml (hunk 7 above)
Comment (ckstettler) [reply .github/stale.yml:44]: Will we know that it was closed by the app?

File: .github/stale.yml (hunk 8 above)
Comment (ckstettler) [reply .github/stale.yml:47]: 30 actions per hour? What does that mean?

File: .github/stale.yml (hunk 9 above)
Comment (ckstettler) [reply .github/stale.yml:50]: yes

File: .github/stale.yml (hunk 7 above)
Comment (tyleroconnell) [reply .github/stale.yml:44]: Yes. Though we won't have a text body if opt not to put a closeComment, the post will come from a bot ([similar to this](https://github.com/nextcloud/server/issues/23485#issuecomment-727833311))

File: .github/stale.yml (hunk 8 above)
Comment (tyleroconnell) [reply .github/stale.yml:47]: Basically that the bot is capped at running 30 operations within an hour. My assumption would be that this is for when folks wanted to rate limit probot

**Comment by ckstettler** [comment]
Did you intend to close this PR?

**Comment by tyleroconnell** [comment]
I didn't, I accidentally deleted this branch while doing clean-up of my personal GitHub repos. I've created #38 with the latest decisions from this branch

**Pull request closed without merging**